pkg/target/silent.go

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2020-present Snowplow Analytics Ltd.
+ * All rights reserved.
+ *
+ * This software is made available by Snowplow Analytics, Ltd.,
+ * under the terms of the Snowplow Limited Use License Agreement, Version 1.1
+ * located at https://docs.snowplow.io/limited-use-license-1.1
+ * BY INSTALLING, DOWNLOADING, ACCESSING, USING OR DISTRIBUTING ANY PORTION
+ * OF THE SOFTWARE, YOU AGREE TO THE TERMS OF SUCH LICENSE AGREEMENT.
+ */
+
+package target
+
+import (
+	"errors"
+	"github.com/snowplow/snowbridge/pkg/models"
+)
+
+// SilentTarget holds a new client for silently acking data 
+type SilentTarget struct{}
+
+// SilentTargetConfigFunction creates an SilentTarget
+func SilentTargetConfigFunction() (*SilentTarget, error) {
+	return &SilentTarget{}, nil
+}
+
+// The SilentTargetAdapter type is an adapter for functions to be used as
+// pluggable components for Silent Target. It implements the Pluggable interface.
+type SilentTargetAdapter func(i interface{}) (interface{}, error)
+
+// Create implements the ComponentCreator interface.
+func (f SilentTargetAdapter) Create(i interface{}) (interface{}, error) {
+	return f(i)
+}
+
+// ProvideDefault implements the ComponentConfigurable interface.
+func (f SilentTargetAdapter) ProvideDefault() (interface{}, error) {
+	return nil, nil
+}
+
+// AdaptSilentTargetFunc returns SilentTargetAdapter.
+func AdaptSilentTargetFunc(f func() (*SilentTarget, error)) SilentTargetAdapter {
+	return func(i interface{}) (interface{}, error) {
+		if i != nil {
+			return nil, errors.New("unexpected configuration input for Silent target")
+		}
+
+		return f()
+	}
+}
+
+// Write pushes all messages to the required target
+// It's just acking data, nothing more
+func (st *SilentTarget) Write(messages []*models.Message) (*models.TargetWriteResult, error) {
+	for _, msg := range messages {
+		if msg.AckFunc != nil {
+			msg.AckFunc()
+		}
+	}
+
+	return models.NewTargetWriteResult(
```

Comment:
Does this change the picture for metrics?

Reply:
Not really, `sent` from target write result is then translated to the [filtered count](https://github.com/snowplow/snowbridge/blob/884ebce34310e128078887f6d56d5020c8f51be8/cmd/cli/cli.go#L243). So what has been successfully filtered (e.g. silently acked or sent to the HTTP target) is treated as filtered in metrics